Jas acq2106 new thread

zack-vii opened the pull request:
reviewed version of #1579

joshStillerman committed:
Fix:  acq2106_423_Nst devices to acq435 threads

In order to test the threading behavior under realistic conditions
we need at least 2 devices.  The acq435st was changed to use a
thread that copies the device object.

Issue: #1563 outlines some potential problems along these lines.

This commit changes to acq2106 to match the new acq435 code so they
can be tested on the SPARC network.

zack-vii committed:
Fix: fixed bugs in acq2106_423st

joshStillerman reviewed (commented):
I have a few more fixes for this before we commit.

zack-vii commented:
i will branch off teh pydevcie and devhelp fix so if it gets merged you can rebase from there and finish your developement

joshStillerman committed:
Fix: change threaded devices use tread subclass

Use a subclass of thread to be the worker for data acquisition.
In that thread class use the TreeNode.copy method to make a
copy of the device object.

zack-vii committed:
Fix: added trig handling

zack-vii commented:
retest this please

zack-vii committed:
Fix: use _mimport like import to get acq400_hapi

zack-vii committed:
Fix: options, return values and acq400_hapi import fixed for acq435st

zack-vii committed:
Fix: dont return anything if success

joshStillerman committed:
Fix:  coeffs and offsets must be write_shot
